api/datasets.go

```diff
@@ -220,31 +276,69 @@ func (h *DatasetHandlers) listHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func (h *DatasetHandlers) getHandler(w http.ResponseWriter, r *http.Request) {
-	args, err := parseGetReqArgs(r, strings.TrimPrefix(r.URL.Path, "/get/"))
-	if err != nil {
-		util.RespondWithError(w, err)
+	params := lib.GetParams{}
+	args := &GetReqArgs{}
+	err := parseProxyParams(r, &params)
+	if err != nil && err.Error() == "bad request" {
+		util.WriteErrResponse(w, http.StatusBadRequest, err)
 		return
 	}
+	if err != nil {
+		args, err = parseGetReqArgs(r, strings.TrimPrefix(r.URL.Path, "/get/"))
+		if err != nil {
+			util.RespondWithError(w, err)
+			return
+		}
+		params = args.Params
+	} else {
+		// TODO(arqu): get rid of this once we move the proxy call above lib
```

Comment:
This is a bit ugly but have to do it to populate the "lost" bits as we don't pass the `GetReqArgs` but only hte `GetParams`